File: bankingtool.client/src/app/dto/response/get-transactions-list-response.ts
export interface GetTransactionsListResponse {
  transactionsList: GetTransactionsListResponseTransactionList[];
  accountInfo: GetTransactionsListResponseAccountInfo;
  cardInfo: GetTransactionsListResponseCardInfo[];
}

export interface GetTransactionsListResponseTransactionList {
  transactionId: number;
  amount: number;
  stageBalance: number;
  transactionDate: Date;
  description: string;
  transactionType: string;
  transactionCategory: string;
  fromAccountId: string;
  toAccountId: string;
}

export interface GetTransactionsListResponseAccountInfo {
  accountNumber: string;
  bankName: string;
  accountHolderName: string;
  balance: number;
  accountType: string;
  accountId: number;
}

export interface GetTransactionsListResponseCardInfo {
  cardType: string;
  cardNumber: string;
  bankName: string;
  holderName: string;
  expirationDate: string;
  cvv: number;
  balanceLimit: number | null;
}

File: BankingTool.Model/Dto/ResponseDto/GetTransactionsListResponse.cs
namespace BankingTool.Model.Dto.Response
{
    public class GetTransactionsListResponse
    {
        public List<GetTransactionsListResponseTransactionList> TransactionsList { get; set; }
        public GetTransactionsListResponseAccountInfo AccountInfo { get; set; }
        public List<GetTransactionsListResponseCardInfo> CardInfo { get; set; }
    }
    public class GetTransactionsListResponseTransactionList
    {
        public int TransactionId { get; set; }
        public int Amount { get; set; }
        public int StageBalance { get; set; }
        public DateTime TransactionDate { get; set; }
        public string Description { get; set; }
        public string TransactionType { get; set; }
        public string TransactionCategory { get; set; }
        public string FromAccountId { get; set; } 
        public string ToAccountId { get; set; }
        public string TransactionTag { get; set; }
    }
    public class GetTransactionsListResponseAccountInfo
    {
        public int AccountId { get; set; }
        public string AccountNumber { get; set; }
        public string BankName { get; set; }
        public string AccountHolderName { get; set; }
        public int Balance { get; set; }
        public string AccountType { get; set; }
    }
    public class GetTransactionsListResponseCardInfo
    {
        public string CardType { get; set; }
        public string CardNumber { get; set; }
        public string BankName { get; set; }
        public string HolderName { get; set; }
        public DateTime ExpirationDate { get; set; }
        public int CVV { get; set; }
        public int? BalanceLimit { get; set; }
    }
}
